server/project.go

```diff
@@ -242,9 +242,13 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to marshal request body for creating webhook for project ID: %d", repositoryCreate.ProjectID)).SetInternal(err)
 			}
 		case vcsPlugin.GitHubCom:
+			webhookHost := s.profile.BackendHost
+			if s.profile.BackendHost == "http://localhost" {
+				webhookHost = fmt.Sprintf("%s:%d", s.profile.BackendHost, s.profile.BackendPort)
+			}
```

Comment:
This is not as elegant as I'd like, but if the `BackendHost` is using the `localhost`, it is very likely a port is needed. Though it is currently only for the sake of integration testing.